### DISCUSSION TITLE
Upgrade gradle version to support java 17 and 21

### DISCUSSION THREAD

**Discussion by LeoColman** [kotest/kotest#3951] [6.0 changes]
[REF](https://github.com/kotest/kotest/issues/3918#issuecomment-2017530627)

For breaking changes I think it is right time to upgrade gradle version to support java 17 and 21!

for 6.0 I think this kind of major change is possible!

[gradle compatibility matrix](https://docs.gradle.org/current/userguide/compatibility.html)

**Comment by LeoColman** [comment]
[REF](https://github.com/kotest/kotest/issues/3918#issuecomment-2019247946)

> For breaking changes I think it is right time to upgrade gradle version to support java 17 and 21!
> 
> for 6.0 I think this kind of major change is possible!
> 
> [gradle compatibility matrix](https://docs.gradle.org/current/userguide/compatibility.html)

If we compile with Java8 does it matter what version of gradle we use ?

**Reply by mirageoasis** [reply]
is it? 
https://github.com/kotest/kotest/issues/3848
when I was working on this issue this project did not support java21

**Comment by mirageoasis** [comment]
found out this project gradle supports java21! now gradle version is 8.6.1 
sorry my mistake :(

**Comment by Kantis** [comment]
We're on latest Gradle version, so I think this can be closed
